src/js/activities/edit.js

```diff
@@ -6,20 +6,134 @@ var app = angular.module("confluente");
+        if (activity.canSubscribe) {
+            for (var i = 0; i < activity.numberOfQuestions; i++) {
+                var options = [];
+                for (var j = 0; j < activity.formOptions[i].length; j++) {
+                    options.push(activity.formOptions[i][j]);
+                }
+                $scope.inputs.push({
+                    fullQuestion: activity.questionDescriptions[i],
+                    type: activity.typeOfQuestion[i],
+                    options: options,
+                    required: (activity.required[i] === 'true')
+                });
+            }
+            $scope.deadline.subscriptionDeadline = activity.subscriptionDeadline;
+        }
+
+        // If not subscription form was submitted initially than add the standard two questions to input
+        if ($scope.inputs.length === 0) {
+            $scope.inputs = [
+                {fullQuestion: 'Name', type: "name", options: [''], required: 'true'},
+                {fullQuestion: 'TU/e email', type: "TU/e email", options: [''], required: 'true'}
+            ];
+        }
     });
 
+    // adds an element to the inputs variable
+    $scope.add = function () {
+        var dataObj = {fullQuestion: '', type: "☰ text", options: ['option 1'], required: ''};
+        $scope.inputs.push(dataObj);
+    };
+
+    // removes last element from inputs variable
+    $scope.remove = function () {
+        if ($scope.inputs.length > 2) {
+            $scope.inputs.pop();
+        }
+    };
+
+    // Adds an option to a multiple choice question
+    $scope.addOption = function (input) {
+        var option = 'option ' + (input.options.length + 1).toString();
+        input.options.push(option);
+    };
+
+    // Remove specific option from a multiple choice question
+    $scope.removeOption = function(inputIndex, optionIndex) {
+        $scope.inputs[inputIndex].options.splice(optionIndex, 1);
+    };
+
+    // Toggles the canSubscribe variable
+    $scope.toggleSubscribe = function() {
+        $scope.activity.canSubscribe = !$scope.activity.canSubscribe;
+    };
+
     $scope.loading = false;
+
     // function called when edit of activity is submitted
     $scope.submit = function () {
         $scope.loading = true;
+
+        // Checks whether required fields are empty
+        $scope.empty = !$scope.activity.name || !$scope.activity.description || !$scope.activity.Organizer;
+
+        if ($scope.activity.canSubscribe) {
+            // format form correctly
+            var allDescriptions = [];
+            var allTypes = [];
+            var allOptions = [];
+            var allRequired = [];
+
+            $scope.inputs.forEach(function (dataObj) {
+                allDescriptions.push(dataObj.fullQuestion);
+                allTypes.push(dataObj.type);
+                var optionString = "";
+                for (var i = 0; i < dataObj.options.length; i++) {
+                    if (i !== 0) optionString += ";";
```

Comment:
(not specifically here) To me, it seems more logical to translate between strings and lists/arrays in the backend, such that the frontend can be more abstract

Reply:
can be done at later stage